src/scenes/AcceleratorRouter/Deliverables/VariableInternalComment.tsx

```diff
@@ -6,23 +6,31 @@ import TextField from '@terraware/web-components/components/Textfield/Textfield'
 import DialogBox from 'src/components/common/DialogBox/DialogBox';
 import Button from 'src/components/common/button/Button';
 import strings from 'src/strings';
-import { VariableWithValues } from 'src/types/documentProducer/Variable';
+import { SectionVariableWithValues, VariableWithValues } from 'src/types/documentProducer/Variable';
+import { VariableValue } from 'src/types/documentProducer/VariableValue';
 
 interface VariableInternalCommentProps {
-  variable: VariableWithValues;
+  variable: VariableWithValues | SectionVariableWithValues;
   update: (internalComment: string) => void;
   editing: boolean;
 }
 
 function VariableInternalComment({ variable, update, editing }: VariableInternalCommentProps) {
-  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
-  const [internalComment, setInternalComment] = useState(
-    variable.variableValues && variable.variableValues.length > 0
-      ? variable.variableValues[0].internalComment || ''
-      : ''
-  );
   const theme = useTheme();
 
+  const variableValues = variable?.variableValues || [];
+
+  // For section variables, multiple variableValues are returned, so we need to find the one with the current ID
+  let variableValue: VariableValue | undefined;
+  if (variable.type === 'Section') {
+    variableValue = (variable?.variableValues || []).find((value) => value.variableId === variable.id);
+  } else {
+    variableValue = variableValues[0];
+  }
```

Comment:
This is good to know, thanks for the comment 💯

Reply:
The whole variables with values that are variable values that may also have values thing gets pretty heady pretty quick 😅